GenderPayGap.WebUI/Views/Viewing/EmployerDetails/Employer.cshtml

```diff
@@ -62,14 +64,48 @@
         </div>
 
         <div class="grid-row">
-            <div class="column-two-thirds">
+            <div class="column-two-thirds ">
                 <div class="grid-row">
                     <div class="column-full">
                         <hr style="margin-top: 0"/>
                         <div class="body" style="margin-top: 2em">
+                            @await Html.PartialModelAsync(new Details
+                            {
+                                Id = "WhoNeedsToReport",
+                                LinkText = "Who is required to report, and the deadlines",
+                                SummaryPartial = "EmployerDetails/WhoNeedsToReport"
+                            })
+                            @{
+                                var years = organisation.GetRecentReports(Global.ShowReportYearCount).Select(report => report.GetReportingPeriod()).ToArray();
+                                var options = new List<SelectItemViewModel>();
+                                foreach (string year in years)
+                                {
+                                    SelectItemViewModel option = new SelectItemViewModel()
+                                    {
+                                        Value = year,
+                                        Text = year,
+                                    };
+                                    options.Add(option);
+                                }
+
+                                Html.GovUkSelect(new SelectViewModel()
+                                {
+                                    Classes = "govuk-select",
+                                    Items = options
+                                });
+                                <div id="filteringArea">
+                                    <input type="hidden" id="selectOptionsFromReports"
+                                           value=@JsonConvert.SerializeObject(
+                                                     organisation.GetRecentReports(
+                                                         Global.ShowReportYearCount)
+                                                         .Select(
+                                                             report => report.GetReportingPeriod())
+                                                         .ToArray())>
+                                </div>
+                            }

```

Comment:
Should this code block have ended at line 90? It feels weird we're writing html straight into a code block.
I also think the `GovUkSelect` might need an `await`